hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/placement/algorithms/SCMContainerPlacementRackAware.java

```diff
@@ -338,17 +541,18 @@ private DatanodeDetails chooseNode(List<DatanodeDetails> excludedNodes,
    * @param favoredNodes - list of favoredNodes. It's a hint. Whether the nodes
    *                     are chosen depends on whether they meet the constrains.
    *                     Can be null.
+   * @param usedNodes - list of the nodes that are already used.
```

Comment:
In this method, we have chosenNodes and usedNodes. Are they not the same thing? Could we have just one of them?

Reply:
Ah, I think I see now why we need both. You have chosen - that is what was select in this call. You have used, which is the existing set of nodes before the client called into the topology to get more nodes.